conf.d/autopair.fish

```diff
@@ -2,7 +2,7 @@ set --global autopair_left "(" "[" "{" '"' "'"
 set --global autopair_right ")" "]" "}" '"' "'"
 set --global autopair_pairs "()" "[]" "{}" '""' "''"
 
-test $fish_key_bindings = fish_default_key_bindings \
+test "$fish_key_bindings" = fish_default_key_bindings \
     && set --local mode default \
     || set --local mode insert
```

Comment:
Is this correct? `mode` will become `insert` if `$fish_key_bindings` is an empty string (which I didn't know was possible).

Reply:
> Can you post the code you're trying to run?

Script: https://github.com/kidonng/fastmac/blob/46cb4c0fba631093b01f45deae0f015442e5eba2/script-mac.sh
Log: https://github.com/kidonng/fastmac/runs/1580192763?check_suite_focus=true#step:4:114

Reply:
> Is this correct? `mode` will become `insert` if `$fish_key_bindings` is an empty string (which I didn't know was possible).

True, but I think there's an alternative fix, which is to define a default variable for `fish_key_bindings`:

```fish
test -z "$fish_key_bindings" && set -l fish_key_bindings fish_default_key_bindings
```

Reply:
Thank you! Can you answer my question too?

Reply:
That would work, but I still don't understand why `fish_key_bindings` is not defined at this point. We should be able to rely on it being defined.

Reply:
Alright, what happens is that, because you can set $fish_key_bindings in config.fish, it's only set in the interactive setup.

So it's not set in the snippets:

- If you don't use the universal variable
- At first start

Reply:
Thank you. So, what would be the best way to proceed here?

Reply:
Hmmm...

So in case you have an empty $fish_key_bindings, reacting to it isn't really going to work.

Of course you can *assume* it's default (emacs-style), but if it's not you're binding the wrong keys.

Ideally you'd make it so this binds keys that make sense in either, if that doesn't work you'll have to ask vi-mode users to pick their own bindings.

(in this case it would probably "just work" if the default mode for emacs-bindings was "insert", which tbh makes more sense anyway - I'll have to have a think about that)

Or arrange for your bindings to be added to fish_user_key_bindings and then check the variable *there*.

Reply:
Thanks, but I'm more confused now. Autopairs works fine in either default or [vi-mode](https://github.com/jorgebucaran/autopair.fish/issues/1#issuecomment-739306158).

Reply:
@faho Thanks, but I'm more confused now. Autopair works fine in either default or [vi-mode](https://github.com/jorgebucaran/autopair.fish/issues/1#issuecomment-739306158).